CRM/Core/DAO/AllCoreTables.php

```diff
@@ -43,7 +43,12 @@ public static function flush(): void {
    *   [EntityName => [table => table_name, class => CRM_DAO_ClassName]][]
    */
   public static function getEntities(): array {
-    return EntityRepository::getEntities();
+    $allEntities = EntityRepository::getEntities();
+    // Filter out entities without a table
+    $tables = EntityRepository::getTableIndex();
+    // Filter out entities without a class
+    $classes = EntityRepository::getClassIndex();
+    return array_intersect_key($allEntities, array_flip($tables), array_flip($classes));
```

Comment:
The above 3 `EntityRepository` functions all read from a cache, so calling them all here isn't going to hurt performance. I assume this array_intersect calculation *will* incur a small performance ding, but these aren't big arrays (typical benchmarking tools run transformations on arrays with millions of items and these just contain a hundred or so), so probably minimal.
Such is the price of backward-compatibility.